api/src/main/java/io/onetable/spi/extractor/SourceClient.java

```diff
@@ -48,6 +49,14 @@ public interface SourceClient<COMMIT> extends Closeable {
    */
   SchemaCatalog getSchemaCatalog(OneTable table, COMMIT commit);
 
+  /**
+   * Extracts the {@link OneSchema} as of the latest state.
```

Comment:
This method requires a `OneTable` object as an argument, which corresponds to the table format at a certain point in time. The method should return the schema that matches the `OneTable` object, not the most recent one.

Reply:
Could you please confirm @the-other-tim-brown this?

Reply:
Yes, that's correct

Reply:
Okay, for the point-in-time schema, we would need the commit/snapshot param. I will make the changes

Reply:
@coded9 the changes required for this are quite straightforward. You can copy the implementation for the `getSchemaCatalog` that already handles this but simply return the schema and not the schema catalog object.